tests/phpunit/tests/user/capabilities.php

```diff
@@ -991,6 +991,29 @@ public function test_add_role() {
 		$this->assertFalse( $wp_roles->is_role( $role_name ) );
 	}
 
+	/**
+	 * Test add_role with implied capabilities grant successfully grants capabilities.
+	 */
+	public function test_add_role_with_single_level_capabilities() {
+		$role_name = 'janitor';
+		add_role(
+			$role_name,
+			'Janitor',
+			array(
+				'level_1',
+				'sweep_floors' => false,
+			)
+		);
+		$this->flush_roles();
+
+		// Assign a user to that role.
+		$id   = self::factory()->user->create( array( 'role' => $role_name ) );
+		$user = new WP_User( $id );
+
+		$this->assertTrue( $user->has_cap( 'level_1' ) );
+		$this->assertFalse( $user->has_cap( 'sweep_floors' ) );
```

Comment:
See above, I'm not sure why one would want to provide capabilities in that format. Either you should use a list of capabilities to grant, or if for some reason you want to include capabilities to set them to `false`, then it makes sense to provide a map of `$cap => $grant`.

In other words, IMO the test should become something like:
```suggestion
		add_role(
			$role_name,
			'Janitor',
			array(
				'level_1',
				'sweep_floors',
			)
		);
		$this->flush_roles();

		// Assign a user to that role.
		$id   = self::factory()->user->create( array( 'role' => $role_name ) );
		$user = new WP_User( $id );

		$this->assertTrue( $user->has_cap( 'level_1' ) );
		$this->assertTrue( $user->has_cap( 'sweep_floors' ) );
```